core/include/cubos/debug_render.hpp

```diff
@@ -1,20 +1,60 @@
 #ifndef CUBOS_DEBUG_RENDER_HPP
 #define CUBOS_DEBUG_RENDER_HPP
 
-#include <cubos/io/window.hpp>
+#include <list>
+#include <mutex>
+
 #include <glm/glm.hpp>
+#include <glm/gtc/quaternion.hpp>
+
+#include <cubos/gl/render_device.hpp>
 
 namespace cubos
 {
     class Debug
     {
     public:
-        static void init(cubos::io::Window* window);
-        static void drawCube(glm::vec3 center, glm::vec3 size, float time, glm::vec3 color = glm::vec3(1));
-        static void drawWireCube(glm::vec3 center, glm::vec3 size, float time, glm::vec3 color = glm::vec3(1));
+        static void init(gl::RenderDevice& renderDevice);
+        static void drawCube(glm::vec3 center, glm::vec3 size, float time, glm::quat rotation = glm::quat(),
+                             glm::vec3 color = glm::vec3(1));
+        static void drawWireCube(glm::vec3 center, glm::vec3 size, float time, glm::quat rotation = glm::quat(),
+                                 glm::vec3 color = glm::vec3(1));
         static void drawSphere(glm::vec3 center, float radius, float time, glm::vec3 color = glm::vec3(1));
         static void drawWireSphere(glm::vec3 center, float radius, float time, glm::vec3 color = glm::vec3(1));
         static void flush(glm::mat4 vp, double deltaT);
+
+    private:
+        static gl::RenderDevice* renderDevice;
+        static gl::ConstantBuffer mvpBuffer;
+        static gl::ShaderBindingPoint mvpBindingPoint, colorBindingPoint;
+        static gl::ShaderPipeline pipeline;
+
+        static gl::RasterState fillRasterState, wireframeRasterState;
+
+        struct DebugDrawObject
```

Comment:
Maybe reorder these methods, fields and type so that types come first, then methods and then static fields.